packages/builder/src/internal/lib/buidler-lib.ts

```diff
@@ -0,0 +1,55 @@
+import debug from "debug";
+
+import { BuidlerRuntimeEnvironment } from "../../types";
+import { BuidlerContext } from "../context";
+import { loadConfigAndTasks } from "../core/config/config-loading";
+import { BuidlerError } from "../core/errors";
+import { ERRORS } from "../core/errors-list";
+import { BUIDLER_PARAM_DEFINITIONS } from "../core/params/buidler-params";
+import { getEnvBuidlerArguments } from "../core/params/env-variables";
+import { Environment } from "../core/runtime-environment";
+
+let ctx: BuidlerContext;
+let env: BuidlerRuntimeEnvironment;
+
+if (BuidlerContext.isCreated()) {
+  ctx = BuidlerContext.getBuidlerContext();
+
+  // The most probable reason for this to happen is that this file was imported
+  // from the config file
+  if (ctx.environment === undefined) {
+    throw new BuidlerError(ERRORS.GENERAL.LIB_IMPORTED_FROM_THE_CONFIG);
+  }
+
+  env = ctx.environment;
+} else {
+  ctx = BuidlerContext.createBuidlerContext();
+
+  const buidlerArguments = getEnvBuidlerArguments(
+    BUIDLER_PARAM_DEFINITIONS,
+    process.env
+  );
+
+  if (buidlerArguments.verbose) {
+    debug.enable("buidler*");
+  }
+
+  const config = loadConfigAndTasks(buidlerArguments);
+
+  // TODO: This is here for backwards compatibility.
+  // There are very few projects using this.
+  if (buidlerArguments.network === undefined) {
+    buidlerArguments.network = config.defaultNetwork;
+  }
```

Comment:
let's remove this condition. we don't need to keep backwards compatibility

Reply:
I prefer to leave small fixes like these for later. I think there should be loads of them. And I thought we won't be removing all network-related source code, only EVM.